Open_LISA_SDK/domain/protocol/client_protocol.py

```diff
@@ -196,6 +199,47 @@ def send_command_and_result_as_json_string(self, id, command):
             raise InvalidCommandException(
                 "command '{}' for instrument {} is not valid: {}".format(command, id, err))
 
+    def send_file(self, file_bytes, file_target_name):
+        log.debug("[LATENCY_MEASURE][INIT][{}]".format('send_file'))
+        ts = time()
+        self._message_protocol.send_msg(COMMAND_SEND_FILE)
+        self._message_protocol.send_msg(file_target_name)
+        self._message_protocol.send_msg(file_bytes, encode=False)
+
+        response = self._message_protocol.receive_msg()
+        te = time()
+        log.debug("[LATENCY_MEASURE][FINISH][{}][ELAPSED={} seconds]".format('send_file', te - ts))
+        log.info("RESPONSE FROM SERVER: {}".format(str(response)))
+
+    def get_file(self, remote_file_name, file_target_name):
+        log.debug("[LATENCY_MEASURE][INIT][{}]".format('get_file'))
+        ts = time()
+        self._message_protocol.send_msg(COMMAND_GET_FILE)
+        self._message_protocol.send_msg(remote_file_name)
+        file_found = str(self._message_protocol.receive_msg())
```

Comment:
Le cambiaría el nombre a esta variable para que no quede confusa la siguiente línea (quizás algo similar a los otros métodos):
```
        response_type = self._message_protocol.receive_msg()
        if self.__is_valid_response(response_type):
```

```
if ERROR_RESPONSE == file_found:  
```
Esa línea es la que me parece que queda rara (además de que no te limitas a solo tener errores de file found, podrias tener cualquier error, representado en un string que te mande el servidor)

Reply:
Quisiste decir `if not self.__is_valid_response(response_type):` no? en ese caso Done.

Reply:
Copié el ejemplo de más arriba, en tu caso estás haciendo como vos decis checkeas que sea error

Reply:
Joya entonces Done